packages/admin-panel/src/VizBuilderApp/components/MetadataForm.js

```diff
@@ -0,0 +1,94 @@
+/*
+ * Tupaia
+ *  Copyright (c) 2017 - 2021 Beyond Essential Systems Pty Ltd
+ */
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { useForm } from 'react-hook-form';
+import { Autocomplete, TextField } from '@tupaia/ui-components';
+import { usePermissionGroups, useProjects } from '../api/queries';
+import { useStore } from '../store';
+
+export const MetadataForm = ({ Header, Body, Footer, onSubmit }) => {
+  const { handleSubmit, register, errors } = useForm();
+  const [state, { setValue, setProject }] = useStore();
+  const { data: projects = [], isLoading: isLoadingProjects } = useProjects();
+  const {
+    data: permissionGroups = [],
+    isLoading: isLoadingPermissionGroups,
+  } = usePermissionGroups();
+
+  // Save the default values here so that they are frozen from the store when the component first mounts
+  const [defaults] = useState(state);
+  const { name, permissionGroup, project, summary } = defaults;
+
+  const doSubmit = data => {
+    setValue('name', data.name);
+    setValue('summary', data.summary);
+    setValue('permissionGroup', data.permissionGroup);
+    setProject(data.project);
+    onSubmit();
+  };
+
+  return (
+    <form onSubmit={handleSubmit(doSubmit)} noValidate>
+      <Header />
+      <Body>
+        <Autocomplete
+          id="project"
+          name="project"
+          label="Project"
+          placeholder="Select Project"
+          defaultValue={project}
+          options={projects.map(p => p.code)}
+          disabled={isLoadingProjects}
+          error={!!errors.project}
+          helperText={errors.project && errors.project.message}
+          inputRef={register({
+            required: 'Required',
+          })}
+        />
```

Comment:
I think we're actually be able to save multi projects for a visualisation now. Hopefully this can be easily changed to use `allowMultipleValues` like in Admin-Panel?

Reply:
hmm that's a good point. Setting multiple projects would make it more difficult to display a list of locations though. I had a similar question to Rohan about whether or not we should allow multiple locations to be set on a viz and he thought it wasn't a priority for now. @rohan-bes - do you think this is a similar feature? In other words, can we ignore setting multiple projects on a viz for now?

Reply:
Might need to touch base with you guys on this on monday to confirm we all have a common understanding. As far as I'm aware, the 'Select Project' input is sort of a stub thing for now, as we don't actually associate a dashboard item / report with a project in any way. Whether a dashboard item can be seen in a project is entirely dependant on what dashboard relations the dashboard item has, and we aren't configuring those in the Viz Builder (as of yet).
What this selector (and the one in the main Viz Builder UI) really achieve is a way for determining which hierarchy to use for the preview data.
Regardless I think having this selector is good, and forcing the user to enter something is nice too, as it is a real business rule and even it isn't truly enforced on a technical level. @chris-bes makes a good point that multiple project dashboard items are supported, and aren't even uncommon, but given that this selector here is sort of a stub I'm okay with it being a single project selector for now 👍 

I think implementing a way to save dashboard relations as part of the viz builder is a pretty high priority next feature, and adding multi project support at that point would be nice 👍 

Happy to chat with yall on this one though to get your thoughts on it?